### DISCUSSION TITLE
Associations not working

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1375] [Q&A] [answered]
As a relatively new CFWheels user, I find the docs on Associations confusing and sketchy in places. As a result I still haven't found a simple outer join to work for me.

Here's a simplified example of my situation. I want to outer join these two tables using phase_id on both to join with a one to one relationship:

**Milestones_List**
  record_id
  milestone_name
  phase_id

**Milestone_Phases**
  record_id
  phase_id
  phase_name

In the **milestones.cfc** config:
  table("Milestones_List");
  hasOne(name="milestonePhases", modelName="phases",  foreignKey="phase_id", joinKey="phase_id");

In the **phases.cfc** config:
  table("Milestone_Phases");
  belongsTo(name="milestoneList", modelName="milestones", foreignKey="phase_id", joinKey="phase_id", joinType="outer");

In my controller method where this is called I have:
  milestones = model("milestones").findAll(include="milestonePhases", returnAs="query");

What I run it, I get:
  ```
Wheels.AssociationNotFound
    An association named milestonePhases could not be found on the milestones model.

  Suggested action
    Setup an association in the config method of the models/Milestones.cfc file and name it milestonePhases. You can use the 
    belongsTo, hasOne or hasMany method to set it up.
```

But it does exist. Docs are vague in places so this is the best I can figure from them. 
What am I doing wrong?

**Comment by cfcoder** [comment, marked as answer]
Just thought of trying a reload of the app. Sure enough, now it's all working. Argh!!

Can someone please state when this is necessary? I knew changes in routes.cfm and possibly other /config directory file changes required a reload but not model changes. Is it all model changes?

**Reply by chapmandu** [reply]
Routes, models and filters are cached. 

In a side note, as a general convention your model files should have singular names. It makes associations easier to read.

**Reply by cfcoder** [reply]
Thanks!
